tests/Functional/Adapter/SpoolConsumerTestCase.php

```diff
@@ -280,6 +280,89 @@ public function shouldNotReturnMessagesToBrokerIfFlushFailedAndRequeueIsFalse():
         self::assertCount(5, $messages);
     }
 
+    /**
+     * @return array
+     */
+    public function providePrefetchAndMessageCount(): array
+    {
+        return [
+            'message amount equal to prefetch count' => [10, 10, 1],
+            'message amount multiple of prefetch count' => [5, 15, 3],
+            'message amount less than prefetch count' => [10, 7, 1],
+            'message amount more than prefetch count' => [10, 13, 2],
+        ];
+    }
+
+    /**
+     * @test
+     * @dataProvider providePrefetchAndMessageCount
+     *
+     * @param int $prefetchCount
+     * @param int $messageCount
+     * @param int $expectedFlushCallTimes
+     *
+     * @throws \Throwable
+     */
+    public function shouldFlushWithZeroReadTimeout(int $prefetchCount, int $messageCount, int $expectedFlushCallTimes): void
+    {
+        $this->publishMessages($messageCount);
+
+        $flushCalledTimes = 0;
+        $this->messageHandler->setFlushCallback(function () use (&$flushCalledTimes) {
+            $flushCalledTimes++;
+        });
+        $handleCalledTimes = 0;
```

Comment:
You use two blocks (flush/handle). For better reading you can separate this blocks with one empty line.